source/common/network/connection_impl.cc

```diff
@@ -654,8 +654,10 @@ void ConnectionImpl::onWriteReady() {
   if (connecting_) {
     int error;
     socklen_t error_size = sizeof(error);
-    RELEASE_ASSERT(socket_->getSocketOption(SOL_SOCKET, SO_ERROR, &error, &error_size).rc_ == 0,
-                   "");
+    Api::SysCallIntResult result =
+        socket_->getSocketOption(SOL_SOCKET, SO_ERROR, &error, &error_size);
+    RELEASE_ASSERT(result.rc_ == 0,
```

Comment:
Should we change this RELEASE_ASSERT to ENVOY_BUG + error handling?

Also, are there other examples where getSocketOption is followed by an ASSERT or RELEASE_ASSERT?

Reply:
The other cases of `getSocketOption` all use error handling.